File: 334.py
class Solution:
    def increasingTriplet(self, nums: list[int]) -> bool:
        a = b = float('inf')
        for n in nums:
            if n <= a:
                a = n
            elif n <= b:
                b = n
            else:
                return True
        return False
    
if __name__ == '__main__':
    s = Solution()
    nums = list(map(int, input().split()))
    print(s.increasingTriplet(nums))

File: cpp/334.cpp
#include<iostream>
#include<vector>
#include<climits>

using namespace std;

class Solution {
public:
    bool increasingTriplet(vector<int>& nums) {
        int a = INT_MAX, b = INT_MAX;
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] <= a)
                a = nums[i];
            else if(nums[i] <= b)
                b = nums[i];
            else 
                return true;
        }
        return false;
    }
};

int main(){
    Solution sol;
    int n;
    cin >> n;
    vector<int> nums(n);
    cout << sol.increasingTriplet(nums) << endl;
    return 0;
}